### DISCUSSION TITLE
18 gennaio 2022 A - esercizio 2

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#3] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/5564cdc1-9ce3-43ee-9c9e-d7ee25065736)

**Comment by Frisbon** [comment]
### SOLUZIONE CORRETTA (SPERO)
Mi viene:
- 2a) $AB, BC, BD$
- 2b) Non é in 3NF perché $D\rightarrow AC$ é un omofobo (D not key or AC not in key)
- 2c) $\rho = ABD,BCA,DA,DC,AE,AF$ 

![image_2024-01-14_09-50-36](https://github.com/sapienzastudentsnetwork/bdi2324/assets/84731341/85010e9a-d8ca-4308-9335-e9235db11e8e)









### SOLUZIONE SBAGLIATA, CICIO PIAGNE (mi corregge) SOTTO
(grazie a lui ho capito, grazie **j**onut)

puoi anche non guardarla

![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/84731341/f9b194ce-6fbc-4189-aa45-69c9af9a2f54)

**Reply by IonutCicio** [reply]
Il modo in cui hai trovato la copertura minima è sbagliato. Consideriamo $F=\set{AB \to CD, BC \to AE, D \to AC, A \to EF}$

1. Separa i determinati _(regola della decomposizione)_

 $F_1=\set{AB \to C, AB \to D, BC \to A, BC \to E, D \to A, D \to C, A \to E, A \to F}$
 
 2. Semplifica i determinanti _(quello che hai fatto te, più o meno)_
 
 $(A)^+\_{F_1} = AF$, $(B)^+\_{F_1} = B$, $(C)^+_{F_1} = C$, in questo caso non possiamo semplificare nessun determinante
 
 3. Elimina le dipendenze ridondanti
 
 $F_1=\set{AB \to C, AB \to D, BC \to A, BC \to E, D \to A, D \to C, A \to E, A \to F}$

- $C \in (AB)^+_{F_1 \setminus \set{AB \to C}} = ABDCEF \implies$ posso levare $AB \to C$
 
$F_2 =\set{AB \to D, BC \to A, BC \to E, D \to A, D \to C, A \to E, A \to F}$

- $D \notin (AB)^+\_{F_2 \setminus \set{AB \to D}} = AEF \implies$ ***non*** posso levare $AB \to D$
- $A \notin (BC)^+\_{F_2 \setminus \set{BC \to A}} = BCE \implies$ ***non*** posso levare $BC \to A$
- $E \in (BC)^+\_{F_2 \setminus \set{BC \to E}} = BCADEF \implies$ posso levare $BC \to E$

$F_3 =\set{AB \to D, BC \to A, D \to A, D \to C, A \to E, A \to F}$

- $A \notin (D)^+\_{F_3 \setminus \set{D \to A}} = DC \implies$ ***non*** posso levare $D \to A$
- $C \notin (D)^+\_{F_3 \setminus \set{D \to C}} = DAEF \implies$ ***non*** posso levare $D \to C$

Lo stesso vale per le ultime due dipendenze, quindi la copertura minima è

$\rho = \set{ ABD, ABC, DA, DC, AE, AF }$

> [!WARNING]
> L'ho fatto di fretta, ci potrebbero essere errori. Non ho fatto gli ultimi 2 step.

**Reply by Frisbon** [reply]
Grazie mille per le tue correzioni Ci, domani lo rifaccio!

**Reply by salicius** [reply]
@Frisbon nel 2a nel caso in cui tutti gli attributi sono determinati e non ne rimane nemmeno uno come faresti il calcolo delle chiavi ? per esempio in questi esercizi sono tutti determinati gli attributi di R:
https://github.com/sapienzastudentsnetwork/basi-di-dati-1/discussions/45
https://github.com/sapienzastudentsnetwork/basi-di-dati-1/discussions/109

**Comment by niccolo-fato** [comment]
![2024-01-13 14 40 30](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/b8cc3865-6cef-46c4-a664-b106b0c0650c)
![2024-01-13 14 40 36](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/b742f201-3eef-4af4-a7f9-a546c7181abf)

**Comment by daromacarlo** [comment]
![Immagine 2024-02-05 120313](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/ce913fe9-ff72-483d-8c45-2689b136166a)
![Immagine 2024-02-05 120353](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/476fdabd-7f89-4f49-aa6f-76a6bd90c25a)
![Immagine 2024-02-05 120404](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/7c9c9730-7ff8-425a-ab7d-d2983eb24ba6)
![Immagine 2024-02-05 120415](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/19da7b63-44d0-443f-beb5-f4383da0e4c6)
![Immagine 2024-02-05 120424](https://github.com/sapienzastudentsnetwork/bdi2324/assets/135441131/e88608e9-859b-4a1b-838d-36530ad0e9f2)
